src/main/java/pl/h2security/service/UserService.java

```diff
@@ -0,0 +1,38 @@
+package pl.h2security.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import pl.h2security.repository.UserRepository;
+import pl.h2security.repository.UserRoleRepository;
+import pl.h2security.user.User;
+import pl.h2security.user.UserRole;
+
+import java.util.Optional;
+
+@Service
+public class UserService {
+
+    private static final String DEFAULT_ROLE = "ROLE_USER";
+    private UserRepository userRepository;
+    private UserRoleRepository roleRepository;
+    private PasswordEncoder passwordEncoder;
+
+
+    @Autowired
+    public UserService(UserRepository userRepository, UserRoleRepository roleRepository,
+                       PasswordEncoder passwordEncoder) {
+        this.userRepository = userRepository;
+        this.roleRepository = roleRepository;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    public void addWithDefaultRole(User user) {
+        Optional<UserRole> defaultRole = roleRepository.findById(1L);
```

Comment:
Do not find role by id. What if suddenly role with id 1 is admin? Find it by the role string.